### DISCUSSION TITLE
What is the point of `checkoutPaymentCreate`?

### DISCUSSION THREAD

**Discussion by cipriancaba** [saleor/saleor#7329] [Q&A]
Why do we need this intermediate step rather than simply calling the `checkoutComplete`?

**Comment by gorjan-mishevski** [comment]
I think in this phase the items that are on the cart are temporarily stored on the backed, in case they are not bought by another user meanwhile. Here you can charge the user and be sure that the items ordered will be there. When you call the checkout complete the order gets finalized and the quantity goes down (if configured).

**Reply by cipriancaba** [reply]
Thanks for the reply, but if that's the case, what is the mechanism to release the stock? Also there is a `token` you can send a part of the request, but that seems to not change anything on the payment side of things since it's simply being returned to the customer in the response. The docs show there's a payment that's being created but not charged (just authorization) but that's simply not the case
